tests/unit/TestResultsService.unitTest.js

```diff
@@ -628,25 +628,75 @@ describe('setExpiryDateAndCertificateNumber', () => {
   })
 
   context('submitted test', () => {
-    it('should set the expiryDate and the certificateNumber for "Annual With Certificate" testTypes with testResult "pass", "fail" or "prs"', () => {
-      testResultsDAOMock.testResultsResponseMock = Array.of(testResultsMockDB[0])
-      const testResultsService = new TestResultsService(testResultsDAOMock)
-      let mockData = testResultsMockDB[0]
-      testResultsDAOMock.numberOfrecords = ''
-      testResultsDAOMock.numberOfScannedRecords = ''
-      mockData.testTypes[2].testResult = ''
-      return testResultsService.setExpiryDateAndCertificateNumber(mockData)
-        .then(response => {
-          const expectedExpiryDate = new Date()
-          expectedExpiryDate.setFullYear(new Date().getFullYear() + 1)
-          expectedExpiryDate.setDate(new Date().getDate() - 1)
-          expect((response.testTypes[0].testExpiryDate).split('T')[0]).to.equal(expectedExpiryDate.toISOString().split('T')[0])
-          expect(response.testTypes[0].certificateNumber).to.equal(response.testTypes[0].testNumber)
-          expect(response.testTypes[1].testExpiryDate).to.equal(undefined)
-          expect(response.testTypes[1].certificateNumber).to.equal(undefined)
-          expect(response.testTypes[2].testExpiryDate).to.equal(undefined)
-          expect(response.testTypes[2].certificateNumber).to.equal(undefined)
-        })
+    context('for psv vehicle type', () => {
+      it('should set the expiryDate and the certificateNumber for "Annual With Certificate" testTypes with testResult "pass", "fail" or "prs"', () => {
+        testResultsDAOMock.testResultsResponseMock = Array.of(testResultsMockDB[0])
```

Comment:
use beforeEach for test initialisation , apparently are failing if existing initialisation pattern is subjected to parallel tests
So below goes into beforeEach(...) just above the it(..)

testResultsDAOMock.testResultsResponseMock = Array.of(testResultsMockDB[0])
        const testResultsService = new TestResultsService(testResultsDAOMock)
       testResultsDAOMock.numberOfrecords = ''		         let mockData = testResultsMockDB[0]
       testResultsDAOMock.numberOfScannedRecords = ''		         testResultsDAOMock.numberOfrecords = ''
       mockData.testTypes[2].testResult = ''		         testResultsDAOMock.numberOfScannedRecords = ''
       return testResultsService.setExpiryDateAndCertificateNumber(mockData)		         mockData.testTypes[2].testResult = ''

Reply:
done